experimental/dds/tree2/src/feature-libraries/default-field-kinds/optionalField.ts

```diff
@@ -489,85 +491,59 @@ export const optionalFieldEditor: OptionalFieldEditor = {
 export function optionalFieldIntoDelta(
 	{ change, revision }: TaggedChange<OptionalChangeset>,
 	deltaFromChild: ToDelta,
-): Delta.Mark[] {
+): Delta.FieldChanges {
+	const delta: Mutable<Delta.FieldChanges> = {};
 	const [_, childChange] = change.childChanges?.find(([changeId]) => changeId === "self") ?? [];
+	if (childChange === undefined && change.fieldChange === undefined) {
+		return delta;
```

Comment:
Isn't this incorrectly ignoring changes to detached nodes?

Reply:
I would say it is correctly doing so. We're not allowing those yet (that will be done as a separate PR).